drivers/adc/adc_nrfx_saadc.c

```diff
@@ -432,11 +445,42 @@ static int check_buffer_size(const struct adc_sequence *sequence,
 	return 0;
 }
 
+static bool has_single_ended(const struct adc_sequence *sequence)
+{
+	return sequence->channels & m_data.single_ended_channels;
+}
+
+static void correct_single_ended(const struct adc_sequence *sequence,
+				 uint8_t active_channels)
+{
+	int8_t channel_id = -1;
+	uint32_t selected_channels = sequence->channels;
+	int16_t *sample_it = sequence->buffer;
+	int16_t *sample_it_end = sample_it + (active_channels * sum_samplings(sequence));
```

Comment:
This is wrong. Only samples from one sampling should be corrected here (that one for which `saadc_irq_handler()` was called). I think you could move the call to `correct_single_ended()` to before that `#if defined(ADC_BUFFER_IN_RAM)` and then here you could get the buffer pointer with just `nrf_saadc_buffer_pointer_get(NRF_SAADC)`.

Reply:
I see, I will update it and test it on the 54H20 which seems to be the only one setting `ADC_BUFFER_IN_RAM` :)